### PR TITLE
Replacing Jobs with Pods in the building/signing features.

### PR DESCRIPTION
**Pull request by ybettan**
We are not using any of the Job's features and we are reconciling the operator artifacts (jobs/pods) anyway, so it is better to reconcile pods directly.

---

Fixes https://github.com/kubernetes-sigs/kernel-module-management/issues/465

### CODE REVIEW

**Comment by ybettan** [comment]
/cc @yevgeny-shnaidman 

@yevgeny-shnaidman @qbarrand @mresvanis Do you think we should also rename the job packages?

File: controllers/hub/managedclustermodule_reconciler.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type ManagedClusterModuleReconciler struct {
 //+kubebuilder:rbac:groups=hub.kmm.sigs.x-k8s.io,resources=managedclustermodules/finalizers,verbs=update
 //+kubebuilder:rbac:groups=work.open-cluster-management.io,resources=manifestworks,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=managedclusters,verbs=get;list;watch
-//+kubebuilder:rbac:groups=batch,resources=jobs,verbs=create;list;watch;delete
+//+kubebuilder:rbac:groups=batch,resources=pods,verbs=create;list;watch;delete
```
Comment (qbarrand) [inline controllers/hub/managedclustermodule_reconciler.go:59]:
```suggestion
//+kubebuilder:rbac:groups=core,resources=pods,verbs=create;list;watch;delete
```

File: controllers/module_reconciler.go [hunk 2]
```diff
@@ -87,7 +86,7 @@ func NewModuleReconciler(
 //+kubebuilder:rbac:groups="core",resources=nodes,verbs=get;list;watch
 //+kubebuilder:rbac:groups="core",resources=secrets,verbs=get;list;watch
 //+kubebuilder:rbac:groups="core",resources=configmaps,verbs=get;list;watch
-//+kubebuilder:rbac:groups="batch",resources=jobs,verbs=create;list;watch;delete
+//+kubebuilder:rbac:groups="batch",resources=pods,verbs=create;list;watch;delete
```
Comment (qbarrand) [inline controllers/module_reconciler.go:89]:
```suggestion
//+kubebuilder:rbac:groups="core",resources=pods,verbs=create;list;watch;delete
```

**Comment by yevgeny-shnaidman** [comment]
> /cc @yevgeny-shnaidman
> 
> @yevgeny-shnaidman @qbarrand @mresvanis Do you think we should also rename the job packages?

yes, otherwise it is confusing. We should rename job package under build and sign, and rename the "jobhelper" file to "podhelper"

File: internal/utils/jobhelper.go [hunk 3]
```diff
@@ -15,154 +15,154 @@ import (
...
+func NewPodHelper(client client.Client) PodHelper {
+	return &podHelper{
 		client: client,
 	}
 }
 
-func (jh *jobHelper) IsJobChanged(existingJob *batchv1.Job, newJob *batchv1.Job) (bool, error) {
-	existingAnnotations := existingJob.GetAnnotations()
-	newAnnotations := newJob.GetAnnotations()
+func (jh *podHelper) IsPodChanged(existingPod *v1.Pod, newPod *v1.Pod) (bool, error) {
```
Comment (qbarrand) [inline internal/utils/jobhelper.go:51]:
```suggestion
func (ph *podHelper) IsPodChanged(existingPod *v1.Pod, newPod *v1.Pod) (bool, error) {
```
Please rename all receivers in this file.

File: internal/sign/job/manager.go [hunk 4]
```diff
@@ -15,46 +16,46 @@ import (
...
 		signer:    signer,
-		jobHelper: jobHelper,
+		podHelper: podHelper,
 		registry:  registry,
 	}
 }
 
-func (jbm *signJobManager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
-	jobs, err := jbm.jobHelper.GetModuleJobs(ctx, modName, namespace, utils.JobTypeSign, owner)
+func (jbm *signPodManager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
```
Comment (qbarrand) [inline internal/sign/job/manager.go:39]:
```suggestion
func (spm *signPodManager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
```
Please rename all receivers in this file.

File: internal/build/job/manager.go [hunk 5]
```diff
@@ -15,46 +16,46 @@ import (
...
 		maker:     maker,
-		jobHelper: jobHelper,
+		podHelper: podHelper,
 		registry:  registry,
 	}
 }
 
-func (jbm *jobManager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
-	jobs, err := jbm.jobHelper.GetModuleJobs(ctx, modName, namespace, utils.JobTypeBuild, owner)
+func (jbm *podManager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
```
Comment (qbarrand) [inline internal/build/job/manager.go:39]:
```suggestion
func (pm *podManager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
```
Please rename all receivers in this file.

File: support/kmm.spec.yaml [hunk 6]
```diff
@@ -92,9 +92,10 @@ spec:
               message: The API deployment does not have any ready replicas.
           - pass:
               message: There is at least one Ready replica of the Deployment.
+    #FIXME: does this one need to be updated manually?
```
Comment (qbarrand) [inline support/kmm.spec.yaml:95]: It does. I would move to https://troubleshoot.sh/docs/analyze/cluster-pod-statuses/ with the appropriate selectors.

**Review by qbarrand — COMMENTED** [review]
Minor suggestions; lgtm otherwise.

File: internal/utils/jobhelper.go (hunk 3 above)
Comment (ybettan) [reply internal/utils/jobhelper.go]: Thanks. Fixed.

File: internal/sign/job/manager.go (hunk 4 above)
Comment (ybettan) [reply internal/sign/job/manager.go]: Thanks. Fixed.

File: internal/build/job/manager.go (hunk 5 above)
Comment (ybettan) [reply internal/build/job/manager.go]: Thanks. Fixed.

File: support/kmm.spec.yaml (hunk 6 above)
Comment (ybettan) [reply support/kmm.spec.yaml]: Done.

**Comment by ybettan** [comment]
> yes, otherwise it is confusing. We should rename job package under build and sign, and rename the "jobhelper" file to "podhelper"

Done.

**Comment by ybettan** [comment]
```
"msg"="problem running manager" "error"="failed to wait for Module caches to sync: timed out waiting for cache to be synced for Kind *v1beta1.Module"
```

Doesn't seems to be related to the PR, re-triggering the CI.

**Pull request merged**